lua/xcodebuild/platform/macos.lua

```diff
@@ -14,7 +14,9 @@ local M = {}
 ---@param callback function|nil
 ---@return number # job id
 function M.launch_app(appPath, callback)
-  local command = { "open", appPath }
+  local appName = util.get_filename(appPath)
+  local executablePath = appPath .. "/Contents/MacOS/" .. appName
```

Comment:
The executable name doesn't always match the app name. App name can be localized while the executable name is constant. The executable name should be taken from `require("xcodebuild.project.config").settings.productName`.

Please pass it as a parameter.